CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -1929,6 +1934,18 @@ protected function processFormSubmission($contactID) {
     // fix currency ID
     $this->_params['currencyID'] = CRM_Core_Config::singleton()->defaultCurrency;
 
+    // CRM-18854
+    if (CRM_Utils_Array::value('adjust_recur_start_date', $this->_values)) {
+      $pledgeBlock = CRM_Pledge_BAO_PledgeBlock::getPledgeBlock($this->_id);
+      if (CRM_Utils_Array::value('start_date', $this->_params) || !CRM_Utils_Array::value('is_pledge_start_date_visible', $pledgeBlock)) {
+        $pledgeStartDate = CRM_Utils_Array::value('start_date', $this->_params, NULL);
+        $this->_params['receive_date'] = CRM_Pledge_BAO_Pledge::getPledgeStartDate($pledgeStartDate, $pledgeBlock);
+        $recurParams = CRM_Pledge_BAO_Pledge::buildRecurParams($this->_params);
+        $this->_params = array_merge($this->_params, $recurParams);
```

Comment:
I only specifically set contributionRecurID here: https://github.com/civicrm/civicrm-core/pull/8558/commits/c4a031d7aea6ddff1011a8ffe4f40e0bac1aa4bd#diff-ecead08fd9d62dd134a679e60a2ba6b4R1112

I could also remove the additional check for is_recur from the payment processor class, if necessary.

I also noticed a couple of notice errors while trying to submit an auto renew membership > Notice: Undefined index: contribution_other_id in CRM_Contribute_Form_Contribution_Confirm->postProcessMembership() (line 1576 of /home/edsel/git/payment-api/CRM/Contribute/Form/Contribution/Confirm.php).
Notice: Undefined variable: paymentResult in CRM_Contribute_Form_Contribution_Confirm->postProcessMembership() (line 1578 of /home/edsel/git/payment-api/CRM/Contribute/Form/Contribution/Confirm.php).

I will submit a separate patch for that.